### ISSUE TITLE
languageSettings/dictionaries not working for "literate haskell"

### ISSUE THREAD

**Issue by Krantz-XRF** [streetsidesoftware/vscode-spell-checker#512]
I would like to enable `haskell`/`latex` dictionaries for language `literate haskell`, and this is in my `settings.json`:

```json
"cSpell.languageSettings": [
  {
    "languageId": "literate haskell",
    "dictionaries": [
        "haskell",
        "latex"
    ]
  }
],
"cSpell.enableFiletypes": [
  "literate haskell"
]
```

Yet in spell checker preferences I don't see `haskell` and `latex` enabled, also words like `newcommand` etc. are reported in editor. Reopening VSCode tried, doesn't help.

![image](https://user-images.githubusercontent.com/17098235/86717038-5ee3e180-c054-11ea-9398-3e6763b8a748.png)

**Comment by Jason3S** [comment]
Hmm. The config reader breaks up:
```
"languageId": "literate haskell"
```

into:
```
"languageId": ["literate", "haskell"]
```

I'll have to look for a backwards compatible fix.
